### PR TITLE
Add string matching

### PR DESCRIPTION
**Pull request by physikerwelt**
* Step one import classes from citeplag

### CODE REVIEW

File: mathosphere-core/src/main/java/com/formulasearchengine/mathosphere/mathpd/text/BasicStringMatcher.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+ * All matches are found, including repetitions after the first match.
+ * <br />
+ * Findings are independent of the capitalization, otherwise the
+ * tokenizer is not worth mentioning.
+ * <p>
+ * Description: <br/>
+ * Basic String Matching finds exact text matches with n-words, including
+ * repetitions after the first match.
+ *
+ * @author Vincent Stange
```
Comment (physikerwelt) [inline mathosphere-core/src/main/java/com/formulasearchengine/mathosphere/mathpd/text/BasicStringMatcher.java]: @vstange can you pleas self assign yourself as reviewer?

File: mathosphere-core/src/main/java/com/formulasearchengine/mathosphere/mathpd/text/BasicStringMatcher.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+     *                   The alphabet in BoyerMoore is too small?
+     */
+    double scoreSimilarity(String text1, String text2) throws Exception {
+
+        final java.util.List<int[]> matches = compare(text1, text2);
+        int sumMatchLength = 0;
+        for (int[] match : matches) {
+            sumMatchLength += match[4];
+        }
+        return Math.min((double) sumMatchLength / text1.length(), 1.0);
```
Comment (physikerwelt) [inline mathosphere-core/src/main/java/com/formulasearchengine/mathosphere/mathpd/text/BasicStringMatcher.java]: I don't understand the idea of the dividing by text1.length()  (it was source.length in https://bitbucket.org/sciplore/citeplag-dev-backend/src/f7823252ccfd1780d24f03afddd986c35142d439/pds-backend-core/src/main/java/org/sciplore/pds/algorithm/impl/BasicStringMatching.java?at=systemupdate&fileviewer=file-view-default#BasicStringMatching.java-116)
And why did it work before without the (double) prefix if you divide two int values?

File: mathosphere-core/src/test/java/com/formulasearchengine/mathosphere/mathpd/text/BasicStringMatcherTest.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        testCases.put(3, "word not a word");
+    }
+
+    @org.junit.Test
+    public void compare() throws Exception {
+        Map<Tuple2<Integer, Integer>, Integer> expectedMatchLengths = new HashMap<>();
+        expectedMatchLengths.put(Tuple2.of(1, 1), 1);
+        expectedMatchLengths.put(Tuple2.of(1, 2), 1);
+        expectedMatchLengths.put(Tuple2.of(1, 3), 0);
+        Function<Tuple2<Integer, Integer>, Integer> f = entry -> {
```
Comment (physikerwelt) [inline mathosphere-core/src/test/java/com/formulasearchengine/mathosphere/mathpd/text/BasicStringMatcherTest.java]: I'm not an expoerts on \lambda expressions... but is there a way to pass the matcher::compare function to the runTest function?

File: mathosphere-core/src/main/java/com/formulasearchengine/mathosphere/mathpd/text/BasicStringMatcher.java (hunk 1 above)
Comment (vstange) [inline mathosphere-core/src/main/java/com/formulasearchengine/mathosphere/mathpd/text/BasicStringMatcher.java]: Okay

File: mathosphere-core/src/main/java/com/formulasearchengine/mathosphere/mathpd/text/BasicStringMatcher.java (hunk 2 above)
Comment (vstange) [inline mathosphere-core/src/main/java/com/formulasearchengine/mathosphere/mathpd/text/BasicStringMatcher.java]: >  I don't understand the idea of the dividing by text1.length() (it was source.length

The score at this point consists of the text coverage of the reference text. The reason is that the reference document should be covered as good as possible and therefore the comparison document was just the working base of the algorithm. This is not a bilateral score.

> And why did it work before without the (double) prefix if you divide two int values?

It doesn't! This algorithm didn't have a proper score value in the beginning and the frontend only wanted to use it as an alternative for the main view. This is an oversight by me.

File: mathosphere-core/src/test/java/com/formulasearchengine/mathosphere/mathpd/text/BasicStringMatcherTest.java (hunk 3 above)
Comment (vstange) [inline mathosphere-core/src/test/java/com/formulasearchengine/mathosphere/mathpd/text/BasicStringMatcherTest.java]: Kind of - yes. If you declare a functional interface, see: http://stackoverflow.com/questions/27872387/can-a-java-lambda-have-more-than-1-parameter

**Pull request merged**